### DISCUSSION TITLE
Auto reply

### DISCUSSION THREAD

**Discussion by MasterBehind** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#61] [General]
Hi,
i just need to modify the code to allow sending welcome meesage each time a customer send a wahtsapp message, 

I believe this is the right code to send the welcome message

                        TextMessageReplyRequest textMessageReplyRequest = new TextMessageReplyRequest();
                        textMessageReplyRequest.Context = new 
                        WhatsappBusiness.CloudApi.Messages.ReplyRequests.TextMessageContext();
                        textMessageReplyRequest.Context.MessageId = textMessage.SingleOrDefault().Id;
                        textMessageReplyRequest.To = textMessage.SingleOrDefault().From;
                        textMessageReplyRequest.Text = new WhatsAppText();
                        textMessageReplyRequest.Text.Body = "Welcome to XYZ company";
                        textMessageReplyRequest.Text.PreviewUrl = false;

                        await _whatsAppBusinessClient.SendTextMessageAsync(textMessageReplyRequest);

The question is, 

1. where should i put this code, i know it should be part of the controler, but where exactly
2. Is they any additional setting i need to make 

Regards

**Comment by gabrieldwight** [comment]
You can put the code in the webhook where you are receiving the message, no additional setting needed

**Reply by MasterBehind** [reply]
Thanks, it did worked now, however, is there any limitation of how many reply buttons i can add to interactive message when using Reply Button?
i noticed it is working when i have up to 3 buttons, after that it throughs error "'(#131009) Parameter value is not valid"

**Reply by MasterBehind** [reply]
if there is limitation, what are the other options i can use in case i have dynamic options to be added that might reach up to 5 or 6

**Reply by gabrieldwight** [reply]
The limit is three buttons for reply button messages

**Comment by MasterBehind** [comment]
And for list type, is there any limit?

**Reply by gabrieldwight** [reply]
I guess 10 or 20 items in a list
